WcaOnRails/config/routes.rb

```diff
@@ -321,6 +321,12 @@
         patch '/wcif' => 'competitions#update_wcif', as: :update_wcif
       end
       get '/records' => "api#records"
+
+      get '/roles' => 'roles#get'
+      post '/roles' => 'roles#post'
+      patch '/roles' => 'roles#patch'
+      delete '/roles' => 'roles#delete'
+      get '/roles/list' => 'roles#list'
```

Comment:
See if `resources` might work for you: https://guides.rubyonrails.org/routing.html#crud-verbs-and-actions

Reply:
I managed to add the 'scope' now. It is working now.